docs/design/satellite-svc-separation.md

```diff
@@ -0,0 +1,93 @@
+# Satellite Service Separation
+
+## Abstract
+
+The goal of the design doc is to describe the work necessary to make the Satellite horizontally scalable.
+
+## Background
+
+Currently all Satellite services run in a single binary in a single container. While this is great for development, this is bad for scaling.
+
+Currently the Satellite is a single binary made of up the following services:
+- version
+- overlay
+- metainfo
+- inspector
+- orders
+- repair (includes checker and repairer)
+- audit
+- garbage collection (GC)
+- accounting (includes tally and rollup service and the live accounting cache)
+- mail
+- console
+- nodestats
+- note: marketing service is currently on pause and is not included in this doc
+- note: kademlia, discovery, and vouchers are being removed and not included in this doc
+
+The Satellite has the following public GRPC endpoints:
+- metainfo
+- nodestats
+- orders
+
+The Satellite has the following privare GRPC endpoints:
+- inspectors
+
+All services (except version) make connections to the masterDB. Five services rely on metainfo service to access the metainfoDB, this includes inspectors, accounting, audit, repair, and garbage collection.
+
+The current items that prevent Satellite horizontal scaling include:
+- live accounting cache is currently stored in memory which cannot be shared across many replicas
+- in-memory databases (i.e. boltDB and sqlite)
+- potentially things that use mutexes, sync.Mutex or sync.RWMutex
```

Comment:
These don't prevent, but indicate that there might be shared state or something transaction hidden.